src/core/authentication.agent.info/agent.info.service.ts

```
@@ -64,6 +64,7 @@ export class AgentInfoService {
       userAgentInfoMetadata.agentID = user.agent.id;
       userAgentInfoMetadata.userID = user.id;
       userAgentInfoMetadata.communicationID = user.communicationID;
+      userAgentInfoMetadata.authId = user.authId ?? undefined;
```

Can it really be `undefined` here?
Sound like an exceptional case to me. Returning an empty string later on will break pretty much everything anyway, better fail early.